### ISSUE TITLE
Context is only preceding context?

### ISSUE THREAD

**Issue by emjotde** [ZurichNLP/ContraPro#2]
Hi again,
Looking through your perl scripts. It seems you only consider preceding context as context, is this correct? I would need to generate my own context differently and then project back to the sentence level to generate scores if my assumptions on context differ.

**Comment by bricksdont** [comment]
Yes, this script only works for preceding context. Would it help if the script could also extract context that follows a focus sentence? i.e. if there was an option to specify a number of following sentences to include?

**Comment by emjotde** [comment]
Yes. At least on the source side I see no reason to not include full context; other than in speech translation you do not have a linear time constraint. 

For context (pun intended), I have a document-level system that is just translating whole documents of up to 1000 tokens as single lines. I don't even count the number of sentences in the context, so unbounded context would be nice. 

My encoder has access to as much context as fits into that limit. My decoder is also producing full translations of up to 1000 tokens. There is no singled out current sentence unless I break the sequence back into pieces. 

There is another question, how to do evaluation correctly in the presence of target context. Including target context from your gold standard seems wrong as this would be quite an unfair advantage. I guess I should only use my own generated target context and infuse your contrast examples.
